components/sensor/ultrasonic/ultrasonic.go

```diff
@@ -198,7 +200,8 @@ func (s *Sensor) measureDistance(ctx context.Context) error {
 	// we calculate the distance to the nearest object based
 	// on the time interval between the sound and its echo
 	// and the speed of sound (343 m/s)
-	distMeters := timeA.Sub(timeB).Seconds() * 343.0 / 2.0
+	secondsElapsed := float64(timeA-timeB) / math.Pow10(6)
```

Comment:
Echoing my earlier comment, I like that the old way used `time.Sub` so you had the units right no matter what. I think the new way is correct right now, but if we ever change the units in the uint32 and forget to change this line, we won't have an easy way to notice that we missed it.